doc/sphinx/how-to/how-to-enable-graphics-core22-on-a-device.md

```diff
@@ -0,0 +1,586 @@
+
+The snap itself must provide the libraries described in 
+[this document](https://mir-server.io/docs/the-graphics-core22-snap-interface#heading--lists--supported-apis).
+Unless you are incredibly space-constrained or in an unusual circumstance, it is a good idea to
+provide **all** of these libraries. Even if your board doesn’t support these libraries, a failure to include
+them will result in applications that would otherwise fall back to a supported API failing to start.
+
+
+`libEGL`, `libGLESv2` and `libgbm` should be provided by the vendor's drivers.
+`libvulkan` *may* be provided by the vendor as well.
+The remaining packages should be available in the Ubuntu archive.
+
+Most of these libraries do **NOT** need to be in a fixed location in the snap.
+However, the following paths **MUST** be provided in a fixed location by your snap:
+
+
+1. libdrm:
+    ```yaml
+    parts:
+      drm-i386:
+        ...
+        organize:
+        # Expected at /libdrm by the `graphics-core22` interface
+        usr/share/libdrm: libdrm
+    ```
+
+
+2. drirc.d:
+    ```yaml
+    parts:
+      dri-i386:
+        organize:
+        # Expected at /drirc.d by the `graphics-core22` interface
+        usr/share/drirc.d: drirc.d
+    ```
+
+
+3. X11:
+    ```yaml
+    parts:
+      x11-i386:
+        organize:
+        # Expected at /X11 by the `graphics-core22` interface
+        usr/share/X11: X11
+    ```
+
+In addition to these paths, the snap **MUST** provide a script at the location
+`bin/graphics-core22-provider-wrapper`.
+This script is invoked with `graphics-core22-provider-wrapper $EXECUTABLE $ARGS...`.
+This script does the following:
+1. It exports any environment variables that are required for the binary 
+to find and use the vendor drivers
+2. It then executes the provided `$EXECUTABLE` with `$ARGS…`.
+
+An example of this script for the nvidia-core22 project is as follows: 
+
+
+```sh
+# https://github.com/canonical/mesa-core22/blob/main/scripts/bin/graphics-core22-provider-wrapper.in
+
+
+#!/bin/bash
+set -euo pipefail
+
+
+# Find the parent directory of the wrapper script. This will be
+# the path of the provider snap in the client. Your files with
+# be found at a path relative to this path.
+SELF="$( cd -- "$(dirname "$0")/.." ; pwd -P )/usr"
+
+
+# mesa-core22 supports multiarch (eg: i386 and amd64 on an amd64 system, or armhf and arm64 on an arm64 system).
+# To do this mesa-core22 needs to supply multiple versions of the libraries, in different paths
+# These paths are architecture-specific, so are filled in at snap build time
+# @ARCH_TRIPLETS@ will be replaced with “i386-linux-gnu x86_64-linux-gnu” when built on amd64
+# and “armhf-linux-gnu arm64-linux-gnu” when built on arm64
+ARCH_TRIPLETS=( @ARCH_TRIPLETS@ )
+
+
+# TODO: Ask Saviq about this one
+if [ "$SNAP_ARCH" == "amd64" ]; then
+  GCONV_PATH=${GCONV_PATH:+$GCONV_PATH:}${SELF}/lib/i386-linux-gnu/gconv
+fi
```

Comment:
@Saviq: Why are we doing this here? Chris and I had very little understanding of this part

Reply:
This is to support i386 runtimes on x86_64 - one of the use cases of this whole thing was games, many of which are i386.

Reply:
I think this example should be simplified to only deal with a single architecture, and a separate section should cover the optional support for other architectures, pointing at the mesa-core22 implementation for reference.

Reply:
So now I just provide the architecture from the snap arch triplet. Let me know if that will suffice. `i386` is mentioned separately